src/main/build.rs

```diff
@@ -390,6 +391,46 @@ fn build_shadow_c(build_common: &ShadowBuildCommon) {
     build.compile("shadow-c");
 }
 
+fn git_version() -> Option<String> {
+    // current git commit version and hash
+    //
+    // `--always` allows us to still get the commit hash and dirty status when tags aren't
+    // available, as sometimes happens when building from a shallow clone
+    let Ok(git_describe) = Command::new("git")
+        .args(["describe", "--always", "--long", "--dirty"])
+        .output()
+    else {
+        return None;
+    };
+
+    if !git_describe.status.success() {
+        return None;
+    }
+
+    // current git commit short date
+    let Ok(git_date) = Command::new("git")
+        .args([
+            "log",
+            "--pretty=format:%ad",
+            "--date=format:%Y-%m-%d--%H:%M:%S",
+            "-n",
+            "1",
+        ])
+        .output()
+    else {
+        return None;
+    };
+
+    if !git_date.status.success() {
+        return None;
+    }
+
+    let git_describe = std::str::from_utf8(&git_describe.stdout).unwrap().trim();
+    let git_date = std::str::from_utf8(&git_date.stdout).unwrap().trim();
```

Comment:
Nit: could maybe use a default string in case we don't get the stdout output as expected. Or maybe we want it to panic if we don't get the stdout output?

Reply:
Changed these to return `None` if git returns invalid utf-8.